src/main/java/lotto/domain/number/CompareLottoNumber.java

```diff
@@ -0,0 +1,72 @@
+package lotto.domain.number;
+
+import java.util.HashMap;
+import java.util.Map;
+import lotto.domain.lotto.Lotto;
+import lotto.domain.lotto.LottoTicket;
+import lotto.domain.lotto.WinningLotto;
+import lotto.domain.result.LottoPrize;
+
+public class CompareLottoNumber {
+
+    private final Map<LottoPrize, Integer> determinePrizeMap = new HashMap<>();
+    private final BonusNumber bonusNumber;
+    private final static int LIMIT_NUMBER = 2;
+    private final static int BONUS_CHECK_NUMBER = 5;
+    private final static int ONE = 1;
+
+    public CompareLottoNumber(
+        LottoTicket lottoTicket,
+        WinningLotto winningLotto,
+        BonusNumber bonusNumber
+    ) {
+        this.bonusNumber = bonusNumber;
+        createDeterminePrize();
+        compareLotto(lottoTicket, winningLotto);
+    }
+
+    public void createDeterminePrize() {
+        for (LottoPrize lottoPrize : LottoPrize.values()) {
+            determinePrizeMap.put(lottoPrize, 0);
+        }
+    }
+
+    private void compareLotto(LottoTicket lottoTicket, WinningLotto winningLotto) {
+        for (Lotto lotto : lottoTicket.getLottoTicket()) {
+            selectSetMap(lotto, winningLotto);
+        }
+    }
+
+    public void setDeterminePrizeMap(int answerCount) {
+        if (LIMIT_NUMBER < answerCount) {
+            determinePrizeMap.put(LottoPrize.findByNumberOfMatches(answerCount),
+                determinePrizeMap.get(LottoPrize.findByNumberOfMatches(answerCount)) + ONE);
```

Comment:
하나의 코드 안에 너무 많은 인자가 들어있어 동작을 파악하기가 어려워요.
```suggestion
            LottoPrize prize = LottoPrize.findByNumberOfMatches(answerCount);
            determinePrizeMap.put(prize, determinePrizeMap.get(prize) + ONE);
```